### ISSUE TITLE
Introduce NSS (as an optional feature)

### ISSUE THREAD

**Issue by leplatrem** [mozilla-services/remote-settings-client#38]
Let's introduce https://nss-crypto.org 

* Add new `nss` feature in Cargo file
* Use `NSS` signature validation if both `openssl` and `nss` are available
* See [rc_crypto](https://github.com/mozilla/application-services/tree/673e700194cfa44ad3c8629f74a2b01844cea7c3/components/support/rc_crypto)

**Comment by ncalexan** [comment]
Oh nice, we're going to do this!  That's great.  I've worked out the in-tree (i.e., in `mozilla-central`) build pieces locally, when the time comes.

I also worked through what's needed to duplicate the `Content-Signature:` verification using NSS primitives.  It's not that challenging, but it's not "out of the box", because [the implementation of ContentSignatureVerifier](https://searchfox.org/mozilla-central/rev/82c04b9cad5b98bdf682bd477f2b1e3071b004ad/security/manager/ssl/ContentSignatureVerifier.cpp) is built at the level of `mozpkix`, which is a level _above_ NSS.  But one can easily work out what's happening (and, in fact, do all the actions with the existing `certutil` command line utility), so it's clear that it can be expressed using just NSS primitives.

Excited to see this happen, and happy to help!
